### PR TITLE
Rework CI

### PR DESCRIPTION
**Pull request by mwestphal**
- Use private actions
- separate CI and coverage in multiple jobs

### CODE REVIEW

File: .github/actions/vtk-install-dep/action.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  using: "composite"
+  steps:
+
+    - name: Recover VTK Short SHA from file
+      if: |
+        steps.cache-vtk.outputs.cache-hit != 'true' &&
+        inputs.vtk_version == 'commit'
+      working-directory: ${{github.workspace}}/dependencies
+      shell: bash
+      run: echo "VTK_SHORT_SHA_OR_TAG=$(<${{ inputs.vtk_sha_file }})" >> $GITHUB_ENV
```
Comment (mwestphal) [inline .github/actions/vtk-install-dep/action.yml:23]: If we store the long sha directly in the file instead, we could skipe the bare repo stuff enterelly.

wdyt @Meakk ?

Comment (Meakk) [reply .github/actions/vtk-install-dep/action.yml:26]: yup, store the long sha then

Comment (mwestphal) [reply .github/actions/vtk-install-dep/action.yml:26]: not working for some reason ? dropping this.

**Comment by mwestphal** [comment]
Coverage result is slightly different has it runs only vtk commit, not all versions.

File: .github/actions/assimp-install-dep/action.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        -DCMAKE_BUILD_TYPE=Release
+        -DCMAKE_INSTALL_PREFIX:PATH=../assimp_install
+        ${{ inputs.os == 'windows-latest' && '-Ax64 -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded' || null }}
+
+    - name: Build ASSIMP
+      if: steps.cache-assimp.outputs.cache-hit != 'true'
+      working-directory: ${{github.workspace}}/dependencies/assimp_build
+      shell: bash
+      run: cmake --build . --parallel 2 --target install --config Release
+
```
Comment (Meakk) [inline .github/actions/assimp-install-dep/action.yml:60]: remove empty line

File: .github/actions/vtk-install-dep/action.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      shell: bash
+      run: echo "VTK_SHA_OR_TAG=$(<${{ inputs.vtk_sha_file }})" >> $GITHUB_ENV
+
+    - name: Copy VTK version to env var
+      if: |
+        steps.cache-vtk.outputs.cache-hit != 'true' &&
+        inputs.vtk_version != 'commit'
+      working-directory: ${{github.workspace}}
+      shell: bash
+      run: echo "VTK_SHA_OR_TAG=${{inputs.vtk_version}}" >> $GITHUB_ENV
```
Comment (Meakk) [inline .github/actions/vtk-install-dep/action.yml:34]: this step looks very similar to the previous one

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -424,10 +172,61 @@ jobs:
         path: ./build/f3d-*
         name: f3d-${{matrix.os}}-${{matrix.vtk_version}}
 
+#----------------------------------------------------------------------------
+# Coverage: Build and test on linux with last VTK with coverage option
+#----------------------------------------------------------------------------
+  coverage:
```
Comment (Meakk) [inline .github/workflows/ci.yml:178]: can we split the regular CI and coverage in two different files?

Comment (mwestphal) [reply .github/workflows/ci.yml:178]: Not if we want a single workflow

File: .github/actions/vtk-install-dep/action.yml (hunk 3 above)
Comment (mwestphal) [reply .github/actions/vtk-install-dep/action.yml:34]: Yes it is different. Check the if condition on top

Comment (Meakk) [reply .github/actions/vtk-install-dep/action.yml:34]: I see

**Review by Meakk — APPROVED** [review]

**Pull request merged**